.github/workflows/differential-shellcheck.yml

```diff
@@ -0,0 +1,26 @@
+name: Differential ShellCheck
+on:
+  pull_request:
+    branches: [main]
+
+permissions:
+  contents: read
```

Comment:
There is already a permissions: on the `jobs/lint` level, so this one seems redundant and unnecessary? I also think `contents` only applies to private repos, but all of cockpit is public. So it seems this could just be dropped?

Reply:
> There is already a permissions: on the `jobs/lint` level, so this one seems redundant and unnecessary?

I usually set permission `contents: read` on top level, in case there would be added another job. But I can merge it with job-level permissions if you would like to.

> I also think `contents` only applies to private repos, but all of cockpit is public. So it seems this could just be dropped?

I don't have much experience with private repos, but I think this also applies to public repos. From [GitHub docs](https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs#assigning-permissions-to-a-specific-job):

* > If you specify the access for any of these scopes, all of those that are not specified are set to none.

So setting permissions `contents: read` indirectly sets everything else to `none`. And then, in each job, you specify precisely which permissions the job requires.

Reply:
@martinpitt Would you prefer the removal of `permissions`?

Reply:
Yes, please. It's not intuitive what happens when there's two `permission:`s. Does the more specific one replace or augment the global one?

Reply:
I have reduced permissions to just:

```yml
permissions:
  contents: read
```

It should be sufficient enough.

Reply:
I am actually surprised that this is actually sufficient -- I would have expected something like `issues: write` to be able to post comments to the PR. But as it worked in [your PR](https://github.com/jamacku/cockpit/pull/1), that' seems to be fine.